internal/provider/resource_site.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/paultyng/go-unifi/unifi"
+)
+
+func resourceSite() *schema.Resource {
+	return &schema.Resource{
+		Description: "`unifi_site` manages Unifi sites",
+
+		Create: resourceSiteCreate,
+		Read:   resourceSiteRead,
+		Update: resourceSiteUpdate,
+		Delete: resourceSiteDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Description: "The description of the site",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "The name of the site",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceSiteCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*client)
+
+	description := d.Get("description").(string)
+
+	resp, err := c.c.CreateSite(context.TODO(), description)
+	if err != nil {
+		return err
+	}
+
+	site := resp[0]
+	d.SetId(site.ID)
+
+	return resourceSiteSetResourceData(&site, d)
+}
+
+func resourceSiteGetResourceData(d *schema.ResourceData) *unifi.Site {
+	return &unifi.Site{
+		ID:          d.Id(),
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+	}
+}
```

Comment:
You can probably remove this since you can only update name/description directly.